paddle/fluid/operators/mkldnn/mul_mkldnn_op.cc

```diff
@@ -0,0 +1,497 @@
+
+    return inner_product_forward::primitive_desc(fc_desc, engine_);
+  }
+
+  mkldnn::inner_product_forward::primitive_desc CreateFcPrimDesc(
+      const mkldnn::memory::desc& input_desc,
+      const mkldnn::memory::desc& weights_desc,
+      const mkldnn::memory::desc& dst_desc) {
+    auto fc_desc = inner_product_forward::desc(prop_kind::forward, input_desc,
+                                               weights_desc, dst_desc);
+
+    return inner_product_forward::primitive_desc(fc_desc, engine_);
+  }
+
+  mkldnn::inner_product_forward::primitive_desc CreateFcPrimDesc(
+      const mkldnn::memory::desc& input_desc,
+      const mkldnn::memory::desc& weights_desc,
+      const mkldnn::memory::desc& dst_desc,
+      const mkldnn::primitive_attr& fc_attr) {
+    auto fc_desc = inner_product_forward::desc(prop_kind::forward, input_desc,
+                                               weights_desc, dst_desc);
+
+    return inner_product_forward::primitive_desc(fc_desc, fc_attr, engine_);
+  }
+
+  virtual mkldnn::memory CreateDstMemory(
+      const mkldnn::inner_product_forward::primitive_desc& fc_prim_desc,
+      const ExecutionContext& ctx, Tensor* output) {
+    auto dst_prim_desc = fc_prim_desc.dst_primitive_desc();
+    auto buffer_size = dst_prim_desc.get_size();
+
+    T* output_data = output->mutable_data<T>(
+        ctx.GetPlace(), ::paddle::memory::Allocator::kDefault, buffer_size);
+    output->set_format((memory::format)dst_prim_desc.desc().data.format);
+    return memory(dst_prim_desc, to_void_cast<T>(output_data));
+  }
+
+ protected:
+  const mkldnn::engine& engine_;
+  boost::optional<memory> bias_;
+  boost::optional<memory> input_;
+  boost::optional<memory> output_;
+  boost::optional<memory> weights_;
+  boost::optional<inner_product_forward> fc_;
+};  // namespace operators
+
+static std::string GetHash(const Tensor* input, const Tensor* weights,
+                           const std::string& suffix) {
+  auto dim2str = [](const DDim& operand_dims) {
+    std::string str = "";
+    for (int i = 0; i < operand_dims.size(); ++i) {
+      str += std::to_string(operand_dims[i]) + "-";
+    }
+    return str;
+  };
+
+  std::string hash = std::to_string((unsigned)input->format()) +
+                     std::to_string((unsigned)input->type()) +
+                     dim2str(weights->dims()) + suffix;
+
+  return hash;
+}
+
+template <typename T, typename K>
+class QuantMulPrimitiveFactory : public MulPrimitiveFactory<T, K> {
+ public:
+  using MulPrimitiveFactory<T, K>::MulPrimitiveFactory;
+
+  virtual inner_product_forward CreateMulPrimitive(
+      const Tensor* input, const Tensor* weights, Tensor* output,
+      const ExecutionContext& ctx) {
+    /* check data format and reorder if need */
+    int x_num_col_dims = ctx.Attr<int>("x_num_col_dims");
+    int y_num_col_dims = ctx.Attr<int>("y_num_col_dims");
+    auto scale_weight = ctx.Attr<std::vector<float>>("scale_y");
```

Comment:
I just want to remind that this is not a fully-connected layer where we have an input and weights, but mul with x and y inputs that are subject to matrix multiplication. It might be helpful to adjust the namings accordingly.

Reply:
okay, will adjust it

Reply:
I tried to modify the variable name from  "input" to "x" and "weight" to "y", but since that some code is shared from FC OP，so reserved the original variable name that in the shared part.

Reply:
done